Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -0,0 +1,140 @@
+        for file in files {
+            // Upon loading a file, the test runner MUST read the schemaVersion field and determine if the test file
+            // can be processed further.
+            guard file.schemaVersion >= Self.minSchemaVersion && file.schemaVersion <= Self.maxSchemaVersion else {
+                throw TestError(
+                    message: "Test file \"\(file.description)\" has unsupported schema version \(file.schemaVersion)"
+                )
+            }
+
+            // If runOnRequirements is specified, the test runner MUST skip the test file unless one or more
+            //  runOnRequirement objects are satisfied.
+            if let requirements = file.runOnRequirements {
+                guard requirements.contains(where: { self.getUnmetRequirement($0) == nil }) else {
+                    fileLevelLog("Skipping tests from file \"\(file.description)\", deployment requirements not met.")
+                    continue
+                }
+            }
+
+            for test in file.tests {
+                // If test.skipReason is specified, the test runner MUST skip this test and MAY use the string value to
+                // log a message.
+                if let skipReason = test.skipReason {
+                    fileLevelLog(
+                        "Skipping test \"\(test.description)\" from file \"\(file.description)\": \(skipReason)."
+                    )
+                    continue
+                }
+
+                // If test.runOnRequirements is specified, the test runner MUST skip the test unless one or more
+                // runOnRequirement objects are satisfied.
+                if let requirements = test.runOnRequirements {
+                    guard requirements.contains(where: { self.getUnmetRequirement($0) == nil }) else {
+                        fileLevelLog(
+                            "Skipping test \"\(test.description)\" from file \"\(file.description)\", " +
+                                "deployment requirements not met."
+                        )
+                        continue
+                    }
+                }
+
+                fileLevelLog("Running test \"\(test.description)\" from file \"\(file.description)\"")
+
+                // If initialData is specified, for each collectionData therein the test runner MUST drop the
+                // collection and insert the specified documents (if any) using a "majority" write concern. If no
+                // documents are specified, the test runner MUST create the collection with a "majority" write concern.
+                // The test runner MUST use the internal MongoClient for these operations.
+                if let initialData = file.initialData {
+                    for collData in initialData {
+                        let db = self.internalClient.db(collData.databaseName)
+                        let coll = db.collection(collData.collectionName)
```

Comment:
it might be easier to just specify the write concern on the collection itself so that if we ever happen to add new statements to this block we won't have to remember to specify it again.

Reply:
done 👍